viscy/representation/evaluation/clustering.py

```diff
@@ -30,6 +33,78 @@ def knn_accuracy(embeddings, annotations, k=5):
     return accuracy
 
 
+def cross_dissimilarity(features: ArrayLike, metric: str = "cosine") -> NDArray:
+    """Dissimilarity/distance between each pair of samples in the features.
```

Comment:
Perhaps `distance_matrix` or `similarity_matrix` is a more intuitive name for the method. The distance is measured by a metric. The metric may be `cosine`, `eucledian`, ...